receiver/redisreceiver/config.md

```diff
@@ -0,0 +1,267 @@
+# Redis Receiver Reference
+
+
+
+### redisreceiver-Config
+
+| Name | Field Info | Default | Docs |
+| ---- | --------- | ------- | ---- |
+| collection_interval |[time-Duration](#time-Duration)| 10s |  |
+| endpoint |string| <no value> |  |
+| transport |string| tcp |  |
+| password |string| <no value> | Optional password. Must match the password specified in the requirepass server configuration option.  |
+| tls |[tls-TLSClientSetting](#tls-TLSClientSetting)| <no value> |  |
+| metrics |[metrics-MetricsSettings](#metrics-MetricsSettings)| <no value> | MetricsSettings provides settings for redisreceiver metrics.  |
+
+### tls-TLSClientSetting
+
+| Name | Field Info | Default | Docs |
+| ---- | --------- | ------- | ---- |
+| ca_file |string| <no value> |  |
+| cert_file |string| <no value> |  |
+| key_file |string| <no value> |  |
+| min_version |string| <no value> |  |
+| max_version |string| <no value> |  |
```

Comment:
These do have defaults; they are just not set on the default configuration. We could work around this with special comments or with struct tags. Could you create a follow up issue for that?

Reply:
When you say that these have defaults, are you referring to the values written by [loadTLSConfig](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/configtls.go#L150)?

Also, there was a bug in configschema in the comments retrieval code which was causing these comments to not get picked up, which I have fixed and pushed to this PR. Perhaps we can indicate how these values get set by putting comments on `TLSClientSettings`?